lib/light-service/organizer/iterate.rb

```diff
@@ -11,12 +12,39 @@ def self.run(organizer, collection_key, steps)
           item_key = collection_key.to_s.singularize.to_sym
           collection.each do |item|
             ctx[item_key] = item
+
             ctx = scoped_reduce(organizer, ctx, steps)
+
+            next unless ctx.key?(:_rollback)
+
+            # Handle Rollback
+            rollback_items(
+              item,
+              organizer,
+              ctx,
+              collection,
+              steps
+            )
           end
 
           ctx
         end
       end
+      # rubocop:enable Metrics/MethodLength
+
+      def self.rollback_items(item, organizer, ctx, collection, steps)
+        reversed_processed_collection = \
+          collection.take_while { |i| i != item }.reverse
```

Comment:
Probably the is a sneaky fact happening invalidating this logic approach: when execution is at https://github.com/adomokos/light-service/pull/206/files#diff-0c78fccee9a0c066001d7b16dd8fe4c02ceebbfa897522ec9f9ecac8d7b4f2f7R39 the Organizer is immediately receiving the first `reduce_rollback` taking the last occurred iteration's actions.

Let's suppose to have

```
1 [A B C]
2 [A B C]
```

and to `fail_with_rollback!` at `2B`. The organizer will immediately receive `reduce_rollback(actions)` having `actions = 2 [A B C]` (if you forgive the pseudocode). Thus, when execution will eventually come back to our `Iterate` class (I'll have to write more on this _eventually_ later), we have to consider that `2 [A B C]` was already rolled back.

This is sneaky because it's a fact, but from the pov of `Iterate` is a matter of faith: it happens in a place it has not vision on.

That said my hypothesis is that `collection.take_while { |i| i != item }` is _taking_ the already rolled back "item" too (already taking in consideration my note at https://github.com/adomokos/light-service/pull/206/files#r601165629)